### DISCUSSION TITLE
How can I add support for Cross-Origin Resource Sharing (CORS) requests in order to fetch assets from Github Enterprise?

### DISCUSSION THREAD

**Discussion by MarcoKouyate** [community/community#135134] [Enterprise] [answered]
### Select Topic Area

Question

### Body

Hello all, 

### Context

In order to use [asset tokens](https://docs.tokens.studio/available-tokens/asset-tokens) in the **Token Studio plugin for Figma**, I need to store images in a web server. The plugin can then fetch those assets by using their URL. I'm considering storing those assets in a private repo with our Github Enterprise workspace. 

### Issue

However, it needs to meet some requirements in order to work with the plugin:
- The assets should not be publicly accessible without auth credentials included in the url 
- The host server need to include **CORS compliant headers** (otherwise the Figma plugin can't fetch the resource)

### My questions

- Is it possible to configure a private repo within Github Enterprise in order to fetch assets with CORS complient requests from Figma's domain?
- Is it possible to include an authentification token on the asset's url in order to keep it from being publicly exposed?

**Comment by DarioSoller** [comment]
We are having the exact same challenge! Our workaround is a private repo on github.com, which is manually synchronized by me to our  Github Enterprise. So I have locally two remote repositories configured.
But we are again searching for a solution to get rid of the proxy repository on github.com, connecting the **Tokens Studio for Figma Plugin** directly to the Github Enterprise.

**Comment by codedbysoumyajit** [comment, marked as answer]
Hey! Great question — this is a pretty specific use case, and unfortunately GitHub-hosted assets (even in private repos) aren’t really set up for secure CORS-based access with auth headers in the URL.

A few points to keep in mind:

1. GitHub doesn’t serve raw assets from private repos in a way that supports CORS headers. Even with GitHub Enterprise, raw files served via raw.githubusercontent.com don’t include CORS headers and aren’t meant for direct access from web clients like Figma plugins.


2. Token-in-URL isn’t supported — GitHub doesn’t allow personal access tokens or OAuth tokens to be passed directly in a file URL for security reasons. So embedding auth tokens in the asset URL (like ?token=...) won’t work either.



- What you can do instead:

   - Host your assets on a CORS-friendly service, like:

   - An S3 bucket (with restricted access and signed URLs)

    - Azure Blob Storage or Google Cloud Storage with temporary access tokens

     - A simple proxy server under your control that handles authentication and adds the required CORS headers


If you really want to automate asset storage from GitHub, you could have a GitHub Action that syncs image files from your private repo to a secure bucket or CDN with proper CORS setup.


Hope that helps! Let me know if you want an example setup for one of those alternatives.
